bin/cheritest/Makefile.cheritest

```diff
@@ -0,0 +1,121 @@
+# $FreeBSD$
+
+.include <src.opts.mk>
+
+SRCS=	cheritest.c
+
+SRCS+=	cheritest_bounds_globals.c					\
+	cheritest_bounds_globals_x.c					\
+	cheritest_bounds_heap.c						\
+	cheritest_bounds_stack.c					\
+	cheritest_fault.c						\
+	cheritest_flag_captured.c					\
+	cheritest_kbounce.c						\
+	cheritest_longjmp.c						\
+	cheritest_printf.c						\
+	cheritest_registers.c						\
+	cheritest_sealcap.c						\
+	cheritest_signal.c						\
+	cheritest_string.c						\
+	cheritest_syscall.c						\
+	cheritest_tls.c							\
+	cheritest_util.c						\
+	cheritest_vm.c							\
+	cheritest_zlib.c
+.if ${MACHINE_ABI:Mpurecap}
+SRCS+=	cheritest_sentries.c
+.endif
+
+CHERITEST_DIR:=	${.PARSEDIR}
+.PATH: ${CHERITEST_DIR}
+
+CFLAGS+=	'-DPROG="${PROG}"'
+
+.if ${MACHINE_ABI:Mpurecap}
+SRCS+=	cheritest_cheriabi.c
+SRCS+=	cheritest_cheriabi_open.c
+.endif
```

Comment:
I wasn't sure which way to go on that one, but I kind of like that all the possible tests are in Makefile.cheritest.

Reply:
I don't really mind which one we use. Both seem fine to me.

Reply:
And I guess since there are various `#ifdef __CHERI_PURE_CAPABILITY__` uses in cheritest it makes sense for the main Makefile.cheritest to key on the ABI rather than have Makefile.purecap opt-in (which would be more akin to using a CHERI_PURECAP_TESTS preprocessor macro).